src/command_modules/azure-cli-network/azure/cli/command_modules/network/custom.py

```diff
@@ -781,24 +829,30 @@ def create_nic(resource_group_name, network_interface_name, subnet, location=Non
                load_balancer_name=None, network_security_group=None,
                private_ip_address=None, private_ip_address_version=IPVersion.ipv4.value,
                public_ip_address=None, virtual_network_name=None):
-    from azure.mgmt.network.models import NetworkInterface
     client = _network_client_factory().network_interfaces
+    NetworkInterface = get_versioned_models(ResourceType.MGMT_NETWORK, 'NetworkInterface')
+    NetworkInterfaceDnsSettings = get_versioned_models(
+        ResourceType.MGMT_NETWORK,
+        'NetworkInterfaceDnsSettings')
     nic = NetworkInterface(location=location, tags=tags, enable_ip_forwarding=enable_ip_forwarding)
     if internal_dns_name_label:
-        from azure.mgmt.network.models import NetworkInterfaceDnsSettings
         nic.dns_settings = NetworkInterfaceDnsSettings(
             internal_dns_name_label=internal_dns_name_label)
     if network_security_group:
         nic.network_security_group = NetworkSecurityGroup(id=network_security_group)
-    ip_config = NetworkInterfaceIPConfiguration(
-        name='ipconfig1',
-        load_balancer_backend_address_pools=load_balancer_backend_address_pool_ids,
-        load_balancer_inbound_nat_rules=load_balancer_inbound_nat_rule_ids,
-        private_ip_allocation_method='Static' if private_ip_address else 'Dynamic',
-        private_ip_address=private_ip_address,
-        private_ip_address_version=private_ip_address_version,
-        subnet=Subnet(id=subnet)
-    )
+
+    ip_config_args = {
+        'name': 'ipconfig1',
+        'load_balancer_backend_address_pools': load_balancer_backend_address_pool_ids,
+        'load_balancer_inbound_nat_rules': load_balancer_inbound_nat_rule_ids,
+        'private_ip_allocation_method': 'Static' if private_ip_address else 'Dynamic',
+        'private_ip_address': private_ip_address,
+        'subnet': Subnet(id=subnet)
+    }
+    if get_api_version(ResourceType.MGMT_NETWORK) in ['2016-09-01']:
```

Comment:
Does this mean we're going to have to manually add '2016-12-01' to everywhere this list occurs to support the next API version?  This would imply that this block applies only to this API version--nothing prior or later.